### PR TITLE
[RLlib] Cleanup examples folder 20: Add example script for new API stack and two separate optimizers (w/ different learning rates).

### PR DESCRIPTION
**Pull request by sven1977**
Cleanup examples folder 20: Add example script for new API stack and two separate optimizers (w/ different learning rates).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/sac/torch/sac_torch_learner.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class SACTorchLearner(DQNRainbowTorchLearner, SACLearner):
     # TODO (simon): Set different learning rates for optimizers.
     @override(DQNRainbowTorchLearner)
     def configure_optimizers_for_module(
-        self, module_id: ModuleID, config: AlgorithmConfig = None, hps=None
+        self, module_id: ModuleID, config: AlgorithmConfig = None
```
Comment (simonsays1980) [inline rllib/algorithms/sac/torch/sac_torch_learner.py:45]: AH nice. This thing was still hanging around there and never used.

File: rllib/examples/learners/classes/separate_vf_lr_and_optimizer_learner.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        assert config.model_config["vf_share_layers"] is False
+
+        # Get all policy-related parameters from the RLModule.
+        pi_params = (
+            # Actor encoder and policy head.
+            self.get_parameters(self.module[module_id].encoder.actor_encoder)
+            + self.get_parameters(self.module[module_id].pi)
+        )
+        # Register the policy optimizer.
+        self.register_optimizer(
```
Comment (simonsays1980) [inline rllib/examples/learners/classes/separate_vf_lr_and_optimizer_learner.py:58]: The nice thing is actually that a user just needs to register this optimizer and does not take care of gradient computations which are way more complex.

File: rllib/examples/learners/separate_vf_lr_and_optimizer.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            # algorithm to use the custom Learner (instead of the default
+            # PPOTorchLearner).
+            learner_class=PPOTorchLearnerWithSeparateVfOptimizer,
+            # We use this simple method here to inject a new setting that our
+            # custom Learner class uses in its `configure_optimizers_for_module`
+            # method. This is convenient and avoids having to subclass `PPOConfig` only
+            # to add a few new settings to it. Within our Learner, we can access this
+            # new setting through:
+            # `self.config.learner_config_dict['lr_vf']`
+            learner_config_dict={"lr_vf": args.lr_vf},
```
Comment (simonsays1980) [inline rllib/examples/learners/separate_vf_lr_and_optimizer.py:118]: As mentioned before. This is not my taste b/c it increases the number attributes in the `AlgorithmConfig` and makes it less readable than simply overriding it.

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: rllib/examples/learners/classes/separate_vf_lr_and_optimizer_learner.py (hunk 2 above)
Comment (sven1977) [reply rllib/examples/learners/classes/separate_vf_lr_and_optimizer_learner.py:58]: Correct! Didn't know this either until I wrote this example. I thought I had to override `compute_gradients`, but the default implementation already takes care of these cases. You only have to override `compute_gradients` if there are shared NN components that need to be "run through" during backward. Like in SAC or DreamerV3.

File: rllib/examples/learners/separate_vf_lr_and_optimizer.py (hunk 3 above)
Comment (sven1977) [reply rllib/examples/learners/separate_vf_lr_and_optimizer.py:118]: I get your point. But I do think this is so much simpler. Users often don't care about the intricate design choices of a complex library like RLlib. So they'll roll their eyes if they have to create an extra subclass of AlgorithmConfig (which is a scary thing to override), just to feed a single setting into their learners.

Instead, they just want things to work as easy as possible.
Note that often, users have a) a very limited set of custom options they need to send in and b) know very well, which are their own settings and which are RLlib settings so there is no real danger of confusing these two.

I make the argument here to a) "make simple things simple" and b) "difficult things possible".

If it helps, I can add comments that explain better how you would do it with a custom AlgorithmConfig override as an alternative.

Also keep in mind that the above is analogous to a custom RLModule, for which you also don't have to subclass any config class of ours. You just send in your custom model config dict.

File: rllib/algorithms/sac/torch/sac_torch_learner.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/sac/torch/sac_torch_learner.py:45]: Yeah, got rid of this.

File: rllib/examples/learners/separate_vf_lr_and_optimizer.py (hunk 3 above)
Comment (grizzlybearg) [reply rllib/examples/learners/separate_vf_lr_and_optimizer.py:118]: Hello @sven1977 @simonsays1980, I'd like to contribute to this discussion as an RLlib user, which might inform your design choices for future RLlib updates. I concur that adding more attributes to `AlgorithmConfig` reduces readability and complicates class modifications for specific use cases. Two years ago, my focus with RLlib usage was on a straightforward interface, without concern for its complexities as @sven1977 has highlighted. Over time, the need for customization became apparent. My point is the necessity for a balance between ease of use and the ability to customize. In essence, I advocate for a) keeping simple tasks uncomplicated, and b) making complex tasks achievable 

> @sven1977 
a) "make simple things simple" and b) "difficult things possible". 

Thank you.

**Pull request merged**